### DISCUSSION TITLE
Custom Bilinear Form

### DISCUSSION THREAD

**Discussion by m-reichelt** [mfem/mfem#2030] [Q&A] [answered]
Hello!
Hopefully I do not ask anything that is answered somewhere in the documentation.
Specifically I want to implement
![image](https://user-images.githubusercontent.com/70210821/106527129-0dff7300-64e7-11eb-91e3-b3e24d681af7.png)
where Y is a fixed value for all testfunctions.
So for most testfunctions it is just the regular stiffness matrix with a minus, but for some it is different.
I looked into nonlinear forms, but I do not think that this is the right way.

Thank you in advance for your help.
Greetings Michael

**Comment by vladotomov** [comment, marked as answer]
One approach is to create a new integrator by taking the code from the existing `DiffusionIntegrator` and write some custom logic around the `el.CalcDShape(ip, dshape);`, by playing with the quadrature point `y-position`.

**Reply by m-reichelt** [reply]
Thank you. I will keep that in mind. I am now trying a different approach for better testability (I directly construct the operator which modifies v(x.y) -> v(x,Y)-v(x,y) and construct its matrix by bilinear form in H1 space. Then I realize the bilinear form with the constructed matrices.
I will keep you updated, if I try out your approach.
